src/plugins/dbus/dbus.c

```diff
@@ -90,31 +82,15 @@ int elektraDbusCommit (Plugin * handle, KeySet * returned, Key * parentKey)
 	ElektraDbusPluginData * pluginData = elektraPluginGetData (handle);
 	ELEKTRA_NOT_NULL (pluginData);
 
-	// because elektraLogchangeGet will always be executed before elektraLogchangeCommit
-	// we know that oldKeys must exist here!
-	KeySet * oldKeys = pluginData->keys;
+	KDB * kdb = elektraPluginGetKdb (handle);
+	const ChangeTrackingContext * context = elektraChangeTrackingGetContextFromKdb (kdb);
 
-	KeySet * addedKeys = ksDup (returned);
-	KeySet * changedKeys = ksNew (0, KS_END);
-	KeySet * removedKeys = ksNew (0, KS_END);
+	// we do a fresh calculation here, because we want the most current diff to avoid false positives as much as possible
```

Comment:
This comment is here because I initially experimented with caching the diffs within the `ChangeTrackingContext`. I've refrained from this as of now. The reason being that we want to eliminate false positives as much as possible, and currently we have no control over when transformations are being applied.

Reply:
We should fixate what transformation plugins are allowed in the decision. A more performant change tracking is much more important than crazy transformation features.

Reply:
For once, I'll be the one to say it: Before we think about performance, we need benchmarks. Maybe not caching the diff is actually not a noticeable performance impact.